java/tools/src/main/java/org/apache/arrow/tools/Integration.java

```diff
@@ -70,23 +73,21 @@ public static void main(String[] args) {
       @Override
       public void execute(File arrowFile, File jsonFile) throws IOException {
         try(
-            BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
-            FileInputStream fileInputStream = new FileInputStream(arrowFile);
-            ArrowReader arrowReader = new ArrowReader(fileInputStream.getChannel(), allocator);) {
+                BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
+                FileInputStream fileInputStream = new FileInputStream(arrowFile);
+                ArrowFileReader arrowReader = new ArrowFileReader(fileInputStream.getChannel(), allocator);) {
           ArrowFooter footer = arrowReader.readFooter();
           Schema schema = footer.getSchema();
           LOGGER.debug("Input file size: " + arrowFile.length());
           LOGGER.debug("Found schema: " + schema);
           try (JsonFileWriter writer = new JsonFileWriter(jsonFile, JsonFileWriter.config().pretty(true));) {
+            VectorSchemaRoot root = new VectorSchemaRoot(footer.getSchema().getFields(), arrowReader.getVectors());
             writer.start(schema);
             List<ArrowBlock> recordBatches = footer.getRecordBatches();
             for (ArrowBlock rbBlock : recordBatches) {
-              try (ArrowRecordBatch inRecordBatch = arrowReader.readRecordBatch(rbBlock);
-                   VectorLoader vectorLoader = new VectorLoader(schema, allocator);) {
-                VectorSchemaRoot root = vectorLoader.getVectorSchemaRoot();
-                vectorLoader.load(inRecordBatch);
-                writer.write(root);
-              }
+              int loaded = arrowReader.loadRecordBatch(rbBlock);
+              root.setRowCount(loaded);
```

Comment:
In this case I was just re-using the VectorSchemaRoot in the unit test for simplicity, because the test checks it, so I wrapped the fields that are loaded into it.
The normal use case would be:
* construct a reader on an input stream
* schema and fields that are loaded are available via getters in the reader - they will read from the input stream when retrieved if not already loaded
* call loadNextBatch or loadRecordBatch (for random access) - returns number of records loaded, and populates fields with those records
* (do something with loaded fields)

It seemed redundant to have the VectorSchemaRoot, when most methods were retrieving the schema from the footer, not the VectorSchemaRoot. If we keep it around, the VectorSchemaRoot would have to be constructed in the reader, since we don't have the schema until something is actually read.